pyvista/core/dataset.py

```diff
@@ -2568,3 +2575,61 @@ def cell_point_ids(self, ind: int) -> List[int]:
         cell = self.GetCell(ind)
         point_ids = cell.GetPointIds()
         return [point_ids.GetId(i) for i in range(point_ids.GetNumberOfIds())]
+
+    def point_is_inside_cell(
+        self, ind: int, point: Union[VectorArray, NumericArray]
+    ) -> Union[int, np.ndarray]:
+        """Return whether a point is inside a cell.
+
+        .. versionadded:: 0.35.0
+
+        Parameters
+        ----------
+        ind : int
+            Cell ID.
+
+        point : Sequence(float) or np.ndarray
+            Coordinates of point to query (length 3) or a ``numpy`` array of ``n``
+            points with shape ``(n, 3)``.
+
+        Returns
+        -------
+        bool or numpy.ndarray
+            Whether point(s) is/are inside cell.
+
+        Examples
+        --------
+        >>> from pyvista import examples
+        >>> mesh = examples.load_hexbeam()
+        >>> mesh.cell_bounds(0)
+        (0.0, 0.5, 0.0, 0.5, 0.0, 0.5)
+        >>> mesh.point_is_inside_cell(0, [0.2, 0.2, 0.2])
+        True
+
+        """
+        if not isinstance(ind, int):
+            raise TypeError(f"ind must be an int, got {type(ind)}")
+
+        if not 0 <= ind < self.n_cells:
+            raise ValueError(f"ind must be >= 0 and < {self.n_cells}, got {ind}")
+
+        point, singular = _coerce_pointslike_arg(point, copy=False)
+
+        cell = self.GetCell(ind)
+        npoints = cell.GetPoints().GetNumberOfPoints()
+
+        closest_point = [0.0, 0.0, 0.0]
+        sub_id = _vtk.mutable(0)
+        pcoords = [0.0, 0.0, 0.0]
+        dist2 = _vtk.mutable(0.0)
+        weights = [0.0] * npoints
+
+        in_cell = np.array(
+            [
+                bool(cell.EvaluatePosition(node, closest_point, sub_id, pcoords, dist2, weights))
```

Comment:
[VTK docs](https://vtk.org/doc/nightly/html/classvtkCell.html#a188d8c01ab1cc03e1fd8b04e3ec97faf) says this can return -1 in case of a "computational problem". We might have to pull out the `bool` conversion, and raise/warn-and-set-to-NaN any broken values. Although the NaN setting would be a but problematic if we're aiming for a bool array.

Reply:
Good catch, let's check and raise if not 0 or 1.  It makes sense to me to have a bool (array) return for this.

Reply:
I'd prefer raising too, my only concern is what happens if someone passes 100 points and one of them has a computational problem? They'd have to remove those points somehow and use the rest.

Reply:
I suppose we should keep the vtk API but it doesn't seem intuitive.    One thought is to put in a Raises section and to suggest to the user to not supply a set of points if they want to catch an error for single points.

Reply:
Hmm, yeah, I think that's a ~good~ least bad compromise. I would also love to keep the bool dtype and the raising. Hopefully computational problems are rare enough that this should only affect few people in very rare cases. We could also mention the recommendation in the error message.

In the off-chance that this starts causing issues downstream we could probably switch from bool to float without breaking existing code.

Reply:
I suspect this will be a very rare corner case, possibly related to parametric cells (I don't know too much about these, but there are several caveats about them in the docs).  vtk docs also mention that this is rare, and there is a very workable solution for those cases, i.e. loop point by point, which shouldn't be that much slower.

Reply:
Implemented a loop to catch `-1` or any other unexpected value.